Dockerfile

```diff
@@ -18,9 +18,16 @@ RUN apt-get update \
     && apt-get install -y --no-install-recommends \
         iptables \
         procps \
+        libcap2-bin \
     && rm -rf /var/lib/apt/lists/* \
     && update-alternatives --set iptables /usr/sbin/iptables-legacy \
     && update-alternatives --set ip6tables /usr/sbin/ip6tables-legacy
 COPY LICENSE /linkerd/LICENSE
 COPY --from=golang /out/linkerd2-proxy-init /usr/local/bin/proxy-init
+
+RUN setcap cap_net_raw,cap_net_admin+eip /usr/sbin/xtables-legacy-multi
+RUN touch /run/xtables.lock && chmod 0666 /run/xtables.lock
+RUN groupadd -r linkerd && useradd --uid 76543 -r -g linkerd linkerd
```

Comment:
@pankajmt  the default uid for proxy init is 2101. I know that you all are using a custom user ID, and this represents a hardcoded value that is incompatible with the default uid. 

I think we'll need to find a way to make this dynamic, if possible

Reply:
@cpretzer do you have a reference for this? i know that the proxy container has a special uid, but i'm not sure if we have one on the proxy-init container as well.

Reply:
Ah, I may be thinking of the proxy uid. I'll double check

Reply:
Happy to make configurable what can be. I thin the proxy-init docker image itself will need a fixed value. 

Following this, we would also need the following changes to allow these to be configured for the security context that is injected for the initContainer by the proxy-init container. 

`diff --git a/charts/linkerd2/values.yaml b/charts/linkerd2/values.yaml
index c7b606ee..69467a3a 100644
--- a/charts/linkerd2/values.yaml
+++ b/charts/linkerd2/values.yaml
@@ -112,6 +112,8 @@ global:
     xtMountPath:
       mountPath: /run
       name: linkerd-proxy-init-xtables-lock
+    runAsNonRoot: false
+    runAsUser: 0
 
   # control plane annotations - do not edit
   createdByAnnotation: linkerd.io/created-by
diff --git a/charts/partials/templates/_proxy-init.tpl b/charts/partials/templates/_proxy-init.tpl
index c818e713..7b3a4108 100644
--- a/charts/partials/templates/_proxy-init.tpl
+++ b/charts/partials/templates/_proxy-init.tpl
@@ -21,11 +21,7 @@ imagePullPolicy: {{.Values.global.proxyInit.image.pullPolicy}}
 name: linkerd-init
 {{ include "partials.resources" .Values.global.proxyInit.resources }}
 securityContext:
-  {{- if .Values.global.proxyInit.closeWaitTimeoutSecs }}
   allowPrivilegeEscalation: true
-  {{- else }}
-  allowPrivilegeEscalation: false
-  {{- end }}
   capabilities:
     add:
     - NET_ADMIN
@@ -44,8 +40,8 @@ securityContext:
   privileged: false
   {{- end }}
   readOnlyRootFilesystem: true
-  runAsNonRoot: false
-  runAsUser: 0
+  runAsNonRoot: {{.Values.global.proxyInit.runAsNonRoot}}
+  runAsUser: {{.Values.global.proxyInit.runAsUser}}
 terminationMessagePolicy: FallbackToLogsOnError
 {{- if or (not .Values.global.cniEnabled) .Values.global.proxyInit.saMountPath }}
 volumeMounts:
@@ -58,5 +54,5 @@ volumeMounts:
 - mountPath: {{.Values.global.proxyInit.saMountPath.mountPath}}
   name: {{.Values.global.proxyInit.saMountPath.name}}
   readOnly: {{.Values.global.proxyInit.saMountPath.readOnly}}
-{{- end -}}  
+{{- end -}}
 {{- end -}}
diff --git a/pkg/charts/linkerd2/values.go b/pkg/charts/linkerd2/values.go
index a425191a..a8f8ac05 100644
--- a/pkg/charts/linkerd2/values.go
+++ b/pkg/charts/linkerd2/values.go
@@ -146,6 +146,8 @@ type (
 		XTMountPath          *VolumeMountPath `json:"xtMountPath"`
 		Resources            *Resources       `json:"resources"`
 		CloseWaitTimeoutSecs int64            `json:"closeWaitTimeoutSecs"`
+		RunAsNonRoot         bool             `json:"runAsNonRoot"`
+		RunAsUser            int64            `json:"runAsUser"`
 	}
 
 	// DebugContainer contains the fields to set the debugging sidecar
`